### PR TITLE
Allow version of ElasticSearch to be specified

### PR DESCRIPTION
**Pull request by leifmadsen**

### CODE REVIEW

**Comment by leifmadsen** [comment]
Not yet tested, so opened as a draft.

**Comment by leifmadsen** [comment]
test

File: deploy/crds/infra.watch_servicetelemetrys_crd.yaml [hunk 1]
```diff
@@ -146,6 +146,9 @@ spec:
                           enabled:
                             description: Enable ElasticSearch as a storage backend for events
                             type: boolean
+                          version:
+                            description: Version of ElasticSearch to deploy. Versions above 7.10.2 are subject to the conditions of Elastic License 2.0 (ELv2) https://www.elastic.co/licensing/elastic-license.
```
Comment (csibbitt) [inline deploy/crds/infra.watch_servicetelemetrys_crd.yaml:150]:
```suggestion
                            description: Version of ElasticSearch to deploy. Elasticsearch licensing has changed as of version 7.11. See https://www.elastic.co/pricing/faq/licensing for details.
```

File: deploy/olm-catalog/service-telemetry-operator/manifests/infra.watch_servicetelemetrys_crd.yaml [hunk 2]
```diff
@@ -127,6 +127,11 @@ spec:
                                 - persistent
                                 type: string
                             type: object
+                          version:
+                            description: Version of ElasticSearch to deploy. Versions
```
Comment (csibbitt) [inline deploy/olm-catalog/service-telemetry-operator/manifests/infra.watch_servicetelemetrys_crd.yaml:131]: Same suggestion here, but the UI makes it a pain to "suggest" it on a multi-line

Comment (leifmadsen) [reply deploy/olm-catalog/service-telemetry-operator/manifests/infra.watch_servicetelemetrys_crd.yaml:131]: This one is generated from the source file you previously commented on.

```
operator-sdk generate bundle --channels unstable --default-channel unstable
```

File: roles/servicetelemetry/templates/manifest_elasticsearch.j2 [hunk 3]
```diff
@@ -68,5 +68,5 @@ spec:
       certificate: {}
   updateStrategy:
     changeBudget: {}
-  version: 7.10.2
+  version: {{ servicetelemetry_vars.backends.ev4ents.elasticsearch.version }}
```
Comment (csibbitt) [inline roles/servicetelemetry/templates/manifest_elasticsearch.j2:71]:
```suggestion
  version: {{ servicetelemetry_vars.backends.events.elasticsearch.version }}
```

**Comment by csibbitt** [comment]
recheck

**Comment by csibbitt** [comment]
test

**Comment by leifmadsen** [comment]
test

File: deploy/olm-catalog/service-telemetry-operator/manifests/service-telemetry-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -37,7 +37,8 @@ metadata:
                       "pvcStorageRequest": "20Gi"
                     },
                     "strategy": "persistent"
-                  }
+                  },
+                  "version": "7.10.2"
```
Comment (csibbitt) [inline deploy/olm-catalog/service-telemetry-operator/manifests/service-telemetry-operator.clusterserviceversion.yaml:41]:
```suggestion
                  "version": "7.16.1"
```

To match https://github.com/infrawatch/service-telemetry-operator/pull/303/files#diff-d511e7d4a1c44b35b99cee004163a2e03c2f63c22077432de49c8f668e7a381fR42

File: deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml [hunk 5]
```diff
@@ -28,6 +28,7 @@ spec:
     events:
       elasticsearch:
         enabled: false
+        version: 7.10.2
```
Comment (csibbitt) [inline deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml:31]:
```suggestion
        version: 7.16.1
```
To match https://github.com/infrawatch/service-telemetry-operator/pull/303/files#diff-d511e7d4a1c44b35b99cee004163a2e03c2f63c22077432de49c8f668e7a381fR42

**Review by csibbitt — COMMENTED** [review]
There is (at least?) one more place this version appears: https://github.com/infrawatch/service-telemetry-operator/blob/master/Jenkinsfile#L48

**Comment by leifmadsen** [comment]
recheck

**Review by csibbitt — APPROVED** [review]

**Comment by leifmadsen** [comment]
Testing by Chris and myself reveals that Elasticsearch can be updated by moving the version value from 7.10.2 to 7.16.1 and the ES Operator will gracefully handle the update in versions.

**Comment by leifmadsen** [comment]
recheck

**Pull request merged**